fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -190,7 +190,7 @@ where
                 msg.gas_limit = fvm_shared::BLOCK_GAS_LIMIT;
             }
 
-            if msg.from == SYSTEM_ACTOR_ADDR {
+            if is_system_addr(&msg.from) {
```

Comment:
Maybe a dumb question, but what happens if a user sends an external transaction with the zero address in the `from`? I guess that it will have no signature and it will be rejected, right? I am just wondering if we are opening the door here to users being able to send free transactions that are executed implicitly by just setting `from = 0x00..`